src/dist/component/components.rs

```diff
@@ -148,12 +149,28 @@ impl<'a> ComponentBuilder<'a> {
 pub struct ComponentPart(pub String, pub PathBuf);
 
 impl ComponentPart {
+    const PATH_SEP_MANIFEST: &str = "/";
+    const PATH_SEP_MAIN: &str = std::path::MAIN_SEPARATOR_STR;
+
     pub(crate) fn encode(&self) -> String {
-        format!("{}:{}", &self.0, &self.1.to_string_lossy())
+        let mut path = self.1.to_string_lossy();
+        if Self::PATH_SEP_MAIN != Self::PATH_SEP_MANIFEST {
+            // Lossy conversion is safe here because we assume that `path` comes from
+            // `ComponentPart::decode()`, i.e. from calling `Path::from()` on a `&str`.
+            path = Cow::Owned(path.replace(Self::PATH_SEP_MAIN, Self::PATH_SEP_MANIFEST));
+        };
+        format!("{}:{path}", self.0)
```

Comment:
It seems not that hard to avoid the extra allocation from concatenating here, maybe try that?

Reply:
(Sorry for the late/slow review.)